chutney/server/src/main/java/com/chutneytesting/campaign/infra/SchedulingCampaignDto.java

```diff
@@ -7,36 +7,64 @@
 
 package com.chutneytesting.campaign.infra;
 
+import static java.util.Collections.emptyList;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 import java.time.LocalDateTime;
+import java.util.Collections;
 import java.util.List;
 
+@JsonPropertyOrder({ "id", "schedulingDate", "environment", "campaignsId", "campaignsTitle", "datasetsId" })
 public class SchedulingCampaignDto {
     public final String id;
-    public final List<Long> campaignsId;
-    public final List<String> campaignsTitle;
     public final LocalDateTime schedulingDate;
     public final String frequency;
+    public final String environment;
+
+    @JsonIgnore
+    public final List<CampaignExecutionRequestDto> campaignExecutionRequestDto;
 
     /**
      * for ObjectMapper only
      **/
+    @JsonCreator
     public SchedulingCampaignDto() {
         id = null;
-        campaignsId = null;
         schedulingDate = null;
-        campaignsTitle = null;
         frequency = null;
+        environment = null;
+        campaignExecutionRequestDto = null;
     }
 
+    @JsonIgnore
     public SchedulingCampaignDto(String id,
-                                 List<Long> campaignsId,
-                                 List<String> campaignsTitle,
                                  LocalDateTime schedulingDate,
-                                 String frequency) {
+                                 String frequency,
+                                 String environment,
+                                 List<CampaignExecutionRequestDto> campaignExecutionRequestDto) {
         this.id = id;
-        this.campaignsId = campaignsId;
-        this.campaignsTitle = campaignsTitle;
         this.schedulingDate = schedulingDate;
         this.frequency = frequency;
+        this.campaignExecutionRequestDto = campaignExecutionRequestDto;
+        this.environment = environment;
+    }
+
+
+    @JsonPropertyOrder
+    public List<Long> getCampaignsId() {
+        return campaignExecutionRequestDto != null ? campaignExecutionRequestDto.stream().map(cer -> cer.campaignId).toList() : emptyList();
+    }
+
+    public List<String> getCampaignsTitle() {
```

Comment:
These getters are not used, are they still relevant ?

Reply:
it's used for json serialization